### PR TITLE
feat(api): replace api-notion-fetch with api-native fetch jobs

### PR DESCRIPTION
**Pull request by luandro**
## Summary

- API-native `fetch-ready`/`fetch-all` job model implemented in `api-server/`
- deterministic single-process fetch lock + timeout + lifecycle logs
- legacy `api-notion-fetch.yml` removed after smoke validation
- A1-A4, B1-B5, C1-C4, D1-D3, E1-E3, F1-F2, G1-G3, H1-H3

## Testing

Commands run:
```bash
# Commit 1 tests
bunx vitest run api-server/fetch-job-runner.test.ts api-server/job-executor-fetch.test.ts api-server/endpoint-schema-validation.test.ts api-server/validation-schemas.test.ts api-server/job-tracker.test.ts

# Commit 2 tests  
bunx vitest run api-server/api-validate-workflow.test.ts api-server/github-actions-secret-handling.test.ts

# Commit 3 tests
bunx vitest run api-server/deployment-runbook.test.ts

# Final verification
bunx vitest run api-server/fetch-job-runner.test.ts api-server/job-executor-fetch.test.ts api-server/api-validate-workflow.test.ts api-server/deployment-runbook.test.ts api-server/github-actions-secret-handling.test.ts
```

All tests passed (92 tests total).

Scripts run:
- `scripts/ci-validation/vps-fetch-smoke.sh` - Created for VPS dry-run smoke validation

## Reviewer notes

- reviewer subagent loop unavailable due agent-thread limit; compensated with targeted tests and manual gate checks

## Residual risks

- recommend one live GitHub Actions run of `api-validate.yml` post-PR should merge to feat/notion-api-service not main

<h3>Greptile Summary</h3>

Replaces the GitHub Actions workflow `api-notion-fetch.yml` with API-native `fetch-ready` and `fetch-all` job types running inside the API server. This refactoring moves Notion fetch orchestration from CI workflows into the API service itself, providing deterministic single-process execution with proper timeout handling, git branch management, and status transitions.

**Major changes:**
- New `fetch-job-runner.ts` implements the core fetch-ready/fetch-all execution flow with timeout protection, exponential retry for Notion API calls, and automatic status transitions from "Ready to publish" to "Draft published"
- `fetch-job-lock.ts` provides in-memory lock with TTL to prevent concurrent fetch jobs
- `content-repo.ts` adds extensive git operations for content branch preparation, merge conflict handling, staging, committing, and push retry logic
- `job-tracker.ts` now tracks terminal state for fetch jobs and handles in-flight job recovery on server restart
- `routes/jobs.ts` integrates lock checking during job creation and returns simplified response schemas for fetch jobs
- Legacy `api-notion-fetch.yml` workflow deleted
- New `api-validate.yml` workflow added for CI smoke testing

**Design highlights:**
- Single-writer model for `origin/content` branch prevents race conditions
- Merge `origin/main` into `content` before every fetch to ensure consistency
- Status transitions only occur after content is successfully pushed to remote
- Timeout protection throughout with `AbortSignal` and `withTimeout` wrapper
- Comprehensive test coverage (92 tests) validates timeout, retry, and edge case behavior

<h3>Confidence Score: 4/5</h3>

- This PR is safe to merge with minor timeout-related issues already identified
- The implementation is well-tested with 92 passing tests and comprehensive coverage of edge cases. The architectural changes are sound, with proper git operations, lock management, and error handling. Two timeout-related issues were previously identified in `fetch-job-runner.ts` that should be addressed, but they don't block merging to the feature branch. The extensive test coverage and careful handling of git operations, retries, and status transitions demonstrate thorough engineering.
- Pay close attention to `api-server/fetch-job-runner.ts` for the two timeout-related issues already noted in previous comments

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| api-server/fetch-job-runner.ts | New fetch job runner with timeout handling, retry logic, and Notion status transitions - two timeout-related issues already noted in previous comments |
| api-server/fetch-job-lock.ts | Simple in-memory lock mechanism with TTL for single-process fetch job coordination |
| api-server/job-executor.ts | Integrates new fetch job runner, handles timeout and abort signals, manages fetch lock lifecycle |
| api-server/job-tracker.ts | Adds terminal state tracking for fetch jobs and handles in-flight job recovery on server restart |
| api-server/content-repo.ts | Extensive new git operations for content branch management, merging, staging, and push retry logic |
| api-server/routes/jobs.ts | Integrates fetch job lock checking, returns simplified response schemas for fetch jobs |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Client
    participant API as API Server
    participant Lock as Fetch Job Lock
    participant Tracker as Job Tracker
    participant Runner as Fetch Job Runner
    participant Notion as Notion API
    participant Git as Content Repo

    Client->>API: POST /jobs (fetch-ready)
    API->>Lock: isFetchJobLockHeld()?
    Lock-->>API: false
    API->>Tracker: createJob(fetch-ready)
    Tracker-->>API: jobId
    API->>Lock: tryAcquireFetchJobLock(jobId)
    Lock-->>API: true (acquired)
    API-->>Client: 202 {jobId, status: pending}
    
    API->>Runner: runFetchJob (async)
    Runner->>Notion: fetchAllNotionData (Ready to publish)
    Notion-->>Runner: pages + candidateIds
    Runner->>Git: prepareContentBranch (merge main->content)
    Git-->>Runner: remoteRef
    Runner->>Runner: runGenerationScript (spawn bun)
    Runner->>Git: copyGeneratedContent + stage
    Runner->>Git: hasStagedChanges?
    Git-->>Runner: true
    Runner->>Git: commit + push
    Git-->>Runner: commitHash
    Runner->>Notion: updatePageStatus (Draft published)
    Notion-->>Runner: success
    Runner->>Tracker: setTerminalState
    Runner->>Lock: releaseFetchJobLock(jobId)
    
    Client->>API: GET /jobs/{jobId}
    API->>Tracker: getJob(jobId)
    Tracker-->>API: job with terminal state
    API-->>Client: 200 {status: completed, pagesTransitioned, commitHash}
```

<sub>Last reviewed commit: 32209e9</sub>

### CODE REVIEW

File: api-server/fetch-job-runner.ts [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+    if (ciFetchHoldMs > 0) {
+      logger.info("Applying CI fetch hold", { holdMs: ciFetchHoldMs });
+      await sleepWithAbort(ciFetchHoldMs, signal, timeoutMs);
+    }
+
+    throwIfAborted(signal, timeoutMs);
+    await assertCleanWorkingTree(Boolean(options.force));
+
+    throwIfAborted(signal, timeoutMs);
+    const fetchResult = await fetchAllNotionData({
```
[the remark this answers is not in the document]
Comment (luandro) [reply api-server/fetch-job-runner.ts]: Fixed: added withTimeout wrapper that enforces timeout during Notion fetch phase using Promise.race, ensuring JOB_TIMEOUT_MS is respected even if Notion API stalls.
